tests/e2e/network/dns_service_connect.go

```diff
@@ -0,0 +1,104 @@
+package network
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"regexp"
+	"time"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	clientset "k8s.io/client-go/kubernetes"
+	testutils "k8s.io/cloud-provider-azure/tests/e2e/utils"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("[Conformance]Service Connection ", func() {
+	var cs clientset.Interface
+	basename := "service"
+	var ns *v1.Namespace
+	var err error
+	var namespacesToDelete []*v1.Namespace
+
+	BeforeEach(func() {
+		By("Creating a kubernetes client")
+		cs, err = testutils.GetClientSet()
+		Expect(err).To(BeNil())
+
+		By("Creating namespace")
+		ns, err = testutils.CreateTestingNS(basename, cs, nil)
+		Expect(err).To(BeNil())
+		namespacesToDelete = append(namespacesToDelete, ns)
+
+	})
+
+	AfterEach(func() {
+		for _, nsToDel := range namespacesToDelete {
+			err = testutils.DeleteNS(cs, nsToDel.Name)
+			Expect(err).NotTo(HaveOccurred())
+		}
+	})
+
+	It("can be done via DNS [Feature: DNS]", func() {
+		serviceName := "dns-service"
+		labels := map[string]string{
+			"app": "dns-server",
+		}
+		ports := []v1.ServicePort{{
+			Port:       8080,
+			TargetPort: intstr.FromInt(8080),
+		}}
+		testutils.Logf("Service: %s\nPort: %v\n", serviceName, ports[0].Port)
+
+		By("Create deployment " + serviceName)
+		cs.Extensions().Deployments(ns.Name).Create(testutils.DefaultDeployment(serviceName, labels))
+
+		By("Create service " + serviceName + " with type LoadBalancer in namespace " + ns.Name)
+		_, err := testutils.CreateLoadBalancerService(cs, serviceName, labels, ns.Name, ports)
+		Expect(err).NotTo(HaveOccurred())
+		testutils.Logf("Service created successfully\n")
+
+		defer func() {
+			By("Cleaning up")
```

Comment:
If directly delete the whole namespace, sometimes(not frequently) it makes bug that the namespace and its pods stop in terminating status

Reply:
Will deleting pods first eliminate the issue?

Reply:
Not sure actually, since I haven't met this issue after then.
Also in upstream test, a resource will be defer deleted once it is created
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/autoscaling/cluster_size_autoscaling.go